src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      const newState = result.length
+        ? { ...result[result.length - 1] } : { ...state };
+
+      for (const key in action.extraData) {
+        newState[key] = action.extraData[key];
+      }
+      result.push(newState);
+    }
+
+    if (action.type === 'removeProperties') {
```

Comment:
Move this word 'removeProperties' to variable and use it here, it will be better